ui-v2/src/components/variables/data-table/data-table.tsx

```diff
@@ -66,35 +80,115 @@ const columns = [
 	}),
 ];
 
+type VariablesDataTableProps = {
+	variables: components["schemas"]["Variable"][];
+	currentVariableCount: number;
+	pagination: PaginationState;
+	onPaginationChange: OnChangeFn<PaginationState>;
+	columnFilters: ColumnFiltersState;
+	onColumnFiltersChange: OnChangeFn<ColumnFiltersState>;
+	searchDebounceMs?: number;
+	sorting: components["schemas"]["VariableSort"];
+	onSortingChange: (sortKey: components["schemas"]["VariableSort"]) => void;
+};
+
 export const VariablesDataTable = ({
 	variables,
-	totalVariableCount,
+	currentVariableCount,
 	pagination,
 	onPaginationChange,
-}: {
-	variables: components["schemas"]["Variable"][];
-	totalVariableCount: number;
-	pagination: PaginationState;
-	onPaginationChange: OnChangeFn<PaginationState>;
-}) => {
+	columnFilters,
+	onColumnFiltersChange,
+	searchDebounceMs = 500,
+	sorting,
+	onSortingChange,
+}: VariablesDataTableProps) => {
+	const initialSearchValue = columnFilters.find(
+		(filter) => filter.id === "name",
+	)?.value as string;
+	const [searchValue, setSearchValue] = useState(initialSearchValue ?? "");
+	const debouncedSearchValue = useDebounce(searchValue, searchDebounceMs);
+
+	const handleNameSearchChange = useCallback(
+		(value: string) => {
+			onColumnFiltersChange((prev) => [
+				...prev.filter((filter) => filter.id !== "name"),
+				{ id: "name", value },
+			]);
+		},
+		[onColumnFiltersChange],
+	);
+
+	const handleTagsSearchChange: React.ChangeEventHandler<HTMLInputElement> &
+		((tags: string[]) => void) = useCallback(
+		(e: string[] | React.ChangeEvent<HTMLInputElement>) => {
+			const tags = Array.isArray(e) ? e : e.target.value;
+
+			onColumnFiltersChange((prev) => [
+				...prev.filter((filter) => filter.id !== "tags"),
+				{ id: "tags", value: tags },
+			]);
+		},
+		[onColumnFiltersChange],
+	);
+
+	useEffect(() => {
+		handleNameSearchChange(debouncedSearchValue);
+	}, [debouncedSearchValue, handleNameSearchChange]);
+
 	const table = useReactTable({
 		data: variables,
 		columns: columns,
 		state: {
 			pagination,
+			columnFilters,
 		},
 		getCoreRowModel: getCoreRowModel(),
 		manualPagination: true,
 		onPaginationChange: onPaginationChange,
-		rowCount: totalVariableCount,
+		onColumnFiltersChange: onColumnFiltersChange,
+		rowCount: currentVariableCount,
 	});
 
 	return (
-		<div className="flex flex-col gap-6 mt-2">
-			<div className="flex flex-row justify-between items-center">
-				<p className="text-sm text-muted-foreground">
-					{totalVariableCount} Variables
-				</p>
+		<div>
+			<div className="grid sm:grid-cols-2 md:grid-cols-6 lg:grid-cols-12 gap-2 pb-4 items-center">
+				<div className="sm:col-span-2 md:col-span-6 lg:col-span-4 order-last lg:order-first">
+					<p className="text-sm text-muted-foreground">
+						{currentVariableCount} Variables
+					</p>
+				</div>
```

Comment:
The filters feel more related to the table than the parent component since I can't imagine a scenario where you'd want the table without the filters, but I'm open to changing this layout as we build more.